brainiak/matnormal/utils.py

```diff
@@ -0,0 +1,73 @@
+import functools  # https://danijar.com/structuring-your-tensorflow-models/
```

Comment:
What is the role of this link? Did the idea come from there? Are you using code from there?

Reply:
It's code from there (or rather, from the gist linked from there). Gist is https://gist.github.com/danijar/8663d3bbfd586bffecf6a0094cd116f2, original website is down but per Wayback machine https://web.archive.org/web/20181116012513/https://danijar.com/structuring-your-tensorflow-models/ it is licensed CC BY-SA 3.0 , citation:

```@misc{hafner2016scopedecorator,
  author = {Hafner, Danijar},
  title = {Structuring Your TensorFlow Models},
  year = {2016},
  howpublished = {Blog post},
  url = {https://danijar.com/structuring-your-tensorflow-models/}
}```

Is that usable if we cite appropriately in that file?

Reply:
I'm afraid we cannot use the reuse code under CC BY-SA 3.0, because that would require we also use CC BY-SA 3.0 for BrainIAK (it's a copylef license). I specifically asked long time ago in the Gist comments for permission to reuse the code under the Apache License 2.0, but got no response:
https://gist.github.com/danijar/8663d3bbfd586bffecf6a0094cd116f2#gistcomment-1982431

Maybe you should ask again, maybe via email, before giving up on reusing the code?

Reply:
I think this is gone now, resolving.